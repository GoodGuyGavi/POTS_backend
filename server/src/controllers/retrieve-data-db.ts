import { IDBModel } from '../commons/types';

const getAllDataDB = (db: IDBModel<any>) => async () => {
	return db.getAll();
};
const getByIDDB = (db: IDBModel<any>) => async id => {
	return db.getById(id);
};

// const getByDeliveryIdDB = (db: IDBModel<any>) => async id => {
//   return db.getByDeliverId(id);
// };

// const getAllByDriverDB = (db: IDBModel<any>) => async id => {
//   return db.getAllByDriver(id);
// };

export { getAllDataDB, getByIDDB };
