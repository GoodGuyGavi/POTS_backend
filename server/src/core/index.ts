import { createUCCreateSupplier } from './_usecases';

export { createUCCreateSupplier };
