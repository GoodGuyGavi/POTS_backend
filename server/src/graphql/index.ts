import { mergeTypes } from 'merge-graphql-schemas';

// import address from "./address";

const typeDefs = [
	//   address,
];

// NOTE: 2nd param is optional, and defaults to false
// Only use if you have defined the same type multiple times in
// different files and wish to attempt merging them together.
export default mergeTypes(typeDefs, { all: true });
