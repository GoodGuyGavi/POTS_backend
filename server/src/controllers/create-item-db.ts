import { IDBModel } from '../commons/types';
import { createUCCreateItem } from '../core/';

const createCreateItemDB = (db: IDBModel<any>) => async i => {
	const createItem = createUCCreateItem();
	const newItem = createItem(i);
	console.log('contorller', newItem);
	return db.insert(newItem);
};

export { createCreateItemDB };
