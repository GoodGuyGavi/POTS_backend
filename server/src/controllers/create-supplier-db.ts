import { IDBModel } from '../commons/types';
import { createUCCreateSupplier } from '../core/_usecases';

const createCreateSupplierDB = (db: IDBModel<any>) => async u => {
	const createSupplier = createUCCreateSupplier();
	const newSupplier = createSupplier(u);
	return db.insert(newSupplier);
};

export { createCreateSupplierDB };
