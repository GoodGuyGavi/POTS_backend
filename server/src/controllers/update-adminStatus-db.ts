import { IDBModel } from '../commons/types';
import { updateUCAdminStatus } from '../core/_usecases';

const updateAdminStatusByIDDB = (db: IDBModel<any>) => async data => {
  const a = await db.getById(data.id);
  const updateAdminStatus = updateUCAdminStatus();
  const newPurchaseOrder = updateAdminStatus(data, a);
  return db.updateById({ ...newPurchaseOrder, id: data.id });
};

export { updateAdminStatusByIDDB };
