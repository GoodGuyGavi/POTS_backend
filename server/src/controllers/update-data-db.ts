import { IDBModel } from '../commons/types';
import { updateUCUser } from '../core/_usecases';

const UpdateByIDDB = (db: IDBModel<any>) => async data => {
	return db.updateById(data);
};

export { UpdateByIDDB };
