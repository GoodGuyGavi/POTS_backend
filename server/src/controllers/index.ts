//User
import { createCreateUserDB } from './create-user-db';
import { DeleteRecordByIDDB } from './delete-data-db';
import { updateUserByIDDB } from './update-user-db';

//Supplier
import { createCreateSupplierDB } from './create-supplier-db';

import { getAllDataDB, getByIDDB } from './retrieve-data-db';

export { getAllDataDB, getByIDDB, createCreateUserDB, DeleteRecordByIDDB, updateUserByIDDB, createCreateSupplierDB };
