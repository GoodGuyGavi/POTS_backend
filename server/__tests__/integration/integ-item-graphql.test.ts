import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import mockData from '../mocks/mock-data';
import * as controllers from '../../src/controllers';
import { constructTestServer } from '../__utils';

const {
  getAllDataDB,
  getByIDDB,
  getAllBySupplierStatusDB,
  getAllByScheduleLineDB,
  DeleteRecordByIDDB,
  createCreateItemDB,
  updateItemByIDDB,
  createCreateScheduleLineDB,
} = controllers;

const itemMock = {
  insert: jest.fn(async input => {
    return {
      id: '1',
      ...input,
      deliveryAddress: 'A1',
    };
  }),
  getById: jest.fn(async id => {
    const filterData = data => {
      if (data.id === id) {
        return data;
      }
    };
    const res = mockData.items.filter(filterData);

    return res[0] || null;
  }),
  getAll: jest.fn(async () => {
    return mockData.items;
  }),
  updateById: jest.fn(async input => {
    return { ...input };
  }),
  deleteById: async id => {
    const filterData = data => {
      if (data.id === id) {
        return data;
      }
    };
    const res = mockData.items.filter(filterData);
    return res[0] || null;
  },
  getAllBySupplierStatus: async id => {},
  getAllByItem: async id => {},
  getAllByScheduleLine: jest.fn(async id => {
    const filterData = data => {
      if (id.includes(data._id)) {
        return data;
      }
    };
    const res = mockData.scheduleLines
      .filter(filterData)
      .map(data => ({ ...data, id: data._id }));
    return res;
  }),
};

const scheduleLinesMock = {
  insert: jest.fn(async input => {
    return { id: '1', ...input };
  }),
  getById: jest.fn(async id => {
    const filterData = data => {
      if (data.id === id) {
        return data;
      }
    };
    const res = mockData.scheduleLines.filter(filterData);

    return res[0] || null;
  }),
  getAll: jest.fn(async () => {
    return mockData.scheduleLines;
  }),
  updateById: jest.fn(async input => {
    return { ...input };
  }),
  deleteById: async id => {
    const filterData = data => {
      if (data.id === id) {
        return data;
      }
    };
    const res = mockData.scheduleLines.filter(filterData);
    return res[0] || null;
  },
  getAllBySupplierStatus: jest.fn(async () => {
    return mockData.scheduleLines;
  }),
  getAllByItem: async id => {},
  getAllByScheduleLine: async id => {},
};

const addressMock = {
  insert: jest.fn(async input => {
    return { id: 'A1', ...input };
  }),
  getAll: jest.fn(async () => {
    return mockData.address;
  }),
  getById: jest.fn(async id => {
    const filterData = data => {
      if (data.id === id) {
        return data;
      }
    };
    const res = mockData.address.filter(filterData);
    return res[0] || null;
  }),
  deleteById: async id => {},
  updateById: async id => {},
  getAllBySupplierStatus: async id => {},
  getAllByItem: async id => {},
  getAllByScheduleLine: async id => {},
};

const { server }: any = constructTestServer({
  context: {
    createItem: createCreateItemDB(itemMock),
    updateItemById: updateItemByIDDB(itemMock),
    deleteItemById: DeleteRecordByIDDB(itemMock),
    getItemById: getByIDDB(itemMock),
    getAllItems: getAllDataDB(itemMock),
    getAllSupplierStatusByItem: getAllBySupplierStatusDB(itemMock),
    getAllScheduleLinesByItem: getAllByScheduleLineDB(itemMock),
    createScheduleLine: createCreateScheduleLineDB(scheduleLinesMock),
    getAddressById: getByIDDB(addressMock),
  },
});

describe('Tests', () => {
  //Queries
  it('should fetch all items', async () => {
    const ITEM_ALL = gql`
      query {
        allItems {
          id
          itemNo
          description
          quantity
          uom
          unitPrice
          currency
          deliveryAddress {
            id
          }
          totalAmount
          supplierStatusItem
          scheduleLine {
            id
            quantity
            uom
            unitPrice
            totalAmount
            deliveryDateAndTime
            deliveryStatus {
              id
              status
              dateCreated
              timeCreated
            }
          }
        }
      }
    `;

    const { query } = createTestClient(server);
    const res = await query({ query: ITEM_ALL });

    expect(res).toMatchSnapshot();
  });

  it('should fetch one item ', async () => {
    const SINGLE_ITEM = gql`
      query i($id: String!) {
        item(id: $id) {
          id
          itemNo
          description
          quantity
          uom
          unitPrice
          currency
          deliveryAddress {
            id
          }
          totalAmount
          supplierStatusItem
          scheduleLine {
            id
            quantity
            uom
            unitPrice
            totalAmount
            deliveryDateAndTime
            deliveryStatus {
              id
              status
              dateCreated
              timeCreated
            }
          }
        }
      }
    `;

    const { query } = createTestClient(server);
    const res = await query({
      query: SINGLE_ITEM,
      variables: { id: '1' },
    });

    expect(res).toMatchSnapshot();
  });

  it('should error when no item', async () => {
    const SINGLE_ITEM = gql`
      query i($id: String!) {
        item(id: $id) {
          id
          itemNo
          description
          quantity
          uom
          unitPrice
          currency
          deliveryAddress {
            id
          }
          totalAmount
          supplierStatusItem
          scheduleLine {
            id
            quantity
            uom
            unitPrice
            totalAmount
            deliveryDateAndTime
            deliveryStatus {
              id
              status
              dateCreated
              timeCreated
            }
          }
        }
      }
    `;

    const { query } = createTestClient(server);
    const res = await query({
      query: SINGLE_ITEM,
      variables: {
        id: '',
      },
    });

    expect(res).toMatchSnapshot();
  });

  //Mutations

  it('update an item', async () => {
    const UPDATE_ITEM = gql`
      mutation i($item: UpdateItemInput!) {
        updateItem(item: $item) {
          id
          supplierStatusItem
        }
      }
    `;

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: UPDATE_ITEM,
      variables: {
        item: {
          id: '1',
          supplierStatusItem: 'Dispatched',
        },
      },
    });

    expect(res).toMatchSnapshot();
  });

  it('update an item supplierStatus', async () => {
    const UPDATE_ITEM = gql`
      mutation i($item: UpdateItemSupplierStatusInput!) {
        updateSupplierItemSupplierStatus(item: $item) {
          id
          supplierStatusItem
        }
      }
    `;

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: UPDATE_ITEM,
      variables: {
        item: {
          id: '1',
          supplierStatusItem: 'Dispatched',
        },
      },
    });

    expect(res).toMatchSnapshot();
  });

  it('delete an item', async () => {
    const DELETE_ITEM = gql`
      mutation i($id: ID!) {
        deleteItem(id: $id) {
          id
        }
      }
    `;

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: DELETE_ITEM,
      variables: { id: '1' },
    });

    expect(res).toMatchSnapshot();
  });
});
