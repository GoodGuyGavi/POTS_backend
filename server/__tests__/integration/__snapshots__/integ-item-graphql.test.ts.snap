// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests delete an item 1`] = `
Object {
  "data": Object {
    "deleteItem": Object {
      "id": "1",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should error when no item 1`] = `
Object {
  "data": Object {
    "item": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should fetch all items 1`] = `
Object {
  "data": Object {
    "allItems": Array [
      Object {
        "currency": "PHP",
        "deliveryAddress": Object {
          "id": "A1",
        },
        "description": "Beef",
        "id": "1",
        "itemNo": "001",
        "quantity": 10,
        "scheduleLine": Array [
          Object {
            "deliveryDateAndTime": "February 25,2020 4:30PM",
            "deliveryStatus": null,
            "id": "1",
            "quantity": 10,
            "totalAmount": 10000,
            "unitPrice": 1000,
            "uom": "kilograms",
          },
        ],
        "supplierStatusItem": null,
        "totalAmount": 10000,
        "unitPrice": 1000,
        "uom": "kilograms",
      },
      Object {
        "currency": "PHP",
        "deliveryAddress": Object {
          "id": "A1",
        },
        "description": "Beef",
        "id": "2",
        "itemNo": "001",
        "quantity": 10,
        "scheduleLine": Array [
          Object {
            "deliveryDateAndTime": "February 25,2020 4:30PM",
            "deliveryStatus": null,
            "id": "2",
            "quantity": 10,
            "totalAmount": 10000,
            "unitPrice": 1000,
            "uom": "kilograms",
          },
        ],
        "supplierStatusItem": null,
        "totalAmount": 10000,
        "unitPrice": 1000,
        "uom": "kilograms",
      },
    ],
  },
  "errors": Array [
    [GraphQLError: getAllSupplierStatusByScheduleLine is not a function],
    [GraphQLError: getAllSupplierStatusByScheduleLine is not a function],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should fetch one item  1`] = `
Object {
  "data": Object {
    "item": Object {
      "currency": "PHP",
      "deliveryAddress": Object {
        "id": "A1",
      },
      "description": "Beef",
      "id": "1",
      "itemNo": "001",
      "quantity": 10,
      "scheduleLine": Array [
        Object {
          "deliveryDateAndTime": "February 25,2020 4:30PM",
          "deliveryStatus": null,
          "id": "1",
          "quantity": 10,
          "totalAmount": 10000,
          "unitPrice": 1000,
          "uom": "kilograms",
        },
      ],
      "supplierStatusItem": null,
      "totalAmount": 10000,
      "unitPrice": 1000,
      "uom": "kilograms",
    },
  },
  "errors": Array [
    [GraphQLError: getAllSupplierStatusByScheduleLine is not a function],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests update an item 1`] = `
Object {
  "data": Object {
    "updateItem": Object {
      "id": "1",
      "supplierStatusItem": "Dispatched",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests update an item supplierStatus 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [ValidationError: Unknown type "UpdateItemSupplierStatusInput". Did you mean "UpdateSupplierStatusInput", "UpdateSupplierStatusItemInput", "UpdateSupplierInput", "SupplierStatusInput", or "UpdateItemInput"?],
    [ValidationError: Cannot query field "updateSupplierItemSupplierStatus" on type "Mutation". Did you mean "updateSupplierStatus", "deleteSupplierStatus", "updateSupplierStatusItem", or "createSupplierStatus"?],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
